sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1325,24 +1325,43 @@ class Analyzer(
      *
      * Note : In this routine, the unresolved attributes are resolved from the input plan's
      * children attributes.
+     *
+     * @param e the expression need to be resolved.
+     * @param q the LogicalPlan use to resolve expression's attribute from.
```

Comment:
it's inaccurate. We resolve the given expression using `q.children`

Reply:
> it's inaccurate. We resolve the given expression using `q.children`

changed